csrc/scheduler/reduction_utils.h

```diff
@@ -56,22 +56,53 @@ void propagateRFactor(
     TensorView* reduction_tv,
     const std::vector<TensorView*>& reduction_tvs);
 
-// Propagate Parallelization from reference TensorView to other TensorViews
-// Parallel types Unroll, Vectorize, and MisalignedVectorize are explicitly
-// handled for tensorviews in cached_inputs and cached_outputs.
-// If reduction_tv and reference_tv shouldn't be unrolled, clear that parallel
-// type. unroll and vectorize are members of ReductionParams
+// Get all cached input/output and shared memory TensorViews that are
+// vectorizable and unrollable.
+//
+// Parameters:
+//   reference_tv: TensorView created during RFactor, used to find vectorizable
+//                 TensorViews.
+//   is_vectorize: Indicates if vectorization is applied in the scheduler.
+//   cached_inputs: Inputs cached in registers or shared memory.
+//   cached_outputs: Outputs cached in registers.
+//   cached_smem_buffer: Shared memory persistent buffers cached in registers.
```

Comment:
Is this tensor cached in registers? Isn't it just cached in shared memory?

Reply:
Yes, they are in registers. The name follows `cached_inputs`. For example, `cached_inputs` means input tvs cached in registers or shared memory. So here `cached_smem_buffer` means smem_buffer cached in registers.
It's a little bit confusing. I modified to `smem_consumers`
```
//   smem_consumers: Consumers of shared memory persistent buffers, they are
//                   register cached Tvs after the shared memory tv.
```

Reply:
Still surprised. Are you saying there's a tensor stored in shared memory and we make a cache of the tensor in registers? Why?

Reply:
We want to vectorized load them from smem to registers to avoid bank conflicts. [Here](https://github.com/NVIDIA/Fuser/blob/973205ef6cc279c692434bded3899ebb7160594b/csrc/scheduler/normalization_utils.cpp#L1267) is the code.

Reply:
It shouldn't be called a cache as there's no reuse.

Reply:
renamed to `smem_consumers`